FactChecker/APIs/KnowledgeGraphAPI/KnowledgeGraphHandler.cs

```diff
@@ -19,49 +21,48 @@ public async Task<List<KnowledgeGraphItem>> GetTriplesBySparQL(string s, int lim
         {
             client.DefaultRequestHeaders.Add("User-Agent", "FactChecker/0.0 (kontakt@magnusaxelsen.dk) generic-library/0.0");
             client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/xml"));
-
             List<KnowledgeGraphItem> triples = new List<KnowledgeGraphItem>();
             try
             {
-                HttpResponseMessage response = await client.GetAsync(knowledgeGraphURL + "?query=SELECT ?r ?t WHERE {wd:" + s + " ?r ?t}ORDER BY ?t limit " + limit);           
+                HttpResponseMessage response = await client.GetAsync(knowledgeGraphURL + "?query=SELECT ?aLabel ?propLabel ?cLabel WHERE { BIND(wd:" + s +" AS ?a). ?a ?b ?c . SERVICE wikibase:label { bd:serviceParam wikibase:language \"en\" . } ?prop wikibase:directClaim ?b . }ORDER BY ?c limit " + limit);
                 if (response.IsSuccessStatusCode)
                 {
-                    XDocument xdoc = XDocument.Parse(response.Content.ReadAsStringAsync().Result);
-                    StringReader sr = new StringReader(xdoc.ToString());
-                    DataSet ds = new DataSet();
-                    ds.ReadXml(sr);
-
-                    foreach (DataTable table in ds.Tables)
+                    var xml = XDocument.Parse(response.Content.ReadAsStringAsync().Result);                  
+                    foreach (var element in xml.Descendants())
                     {
-                        string rSplit = "";
-                        string tSplit = "";
-                        foreach (DataRow row in table.Rows)
+                        if(element.Name == "{http://www.w3.org/2005/sparql-results#}result")
```

Comment:
Kan vi lave det der {http://www.w3.org/2005/sparql-results#}result") om til variabel i stedet for? så bliver det nemmere at ændre i fremtiden perhaps